packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+Clients, at least initially, will be expected to create mechanisms to restrict the amount of data coming to the client-side.
+
+#### _Out of Scope of Initial Release_
+
+The following are features that we intend to prioritize eventually after the initial release:
+
+-   Hierarchy and Grouping together
+-   Custom expanded row content
+    -   An example would be showing a sub-tree when you expand a row
+-   Editable cells
+-   APIs around pagination/infinite scrolling
+-   High-level configuration APIs
+    -   Allow for easy getting/setting of total configurable state, which is useful for maintaining state of the table as a user navigates to other pages in an app.
+-   Filter on any text match across all cells
+-   Selection configuration.
+-   Custom header templates
+-   Custom row templates
+
+### Accessibility
+
+The table will provide the necessary means for keyboard interactions and navigation through the cells, as well as expected ARIA labels/roles for the various elements.
+
+### Risks and Challenges
+
+We will be leveraging a couple of third-party libraries, namely [TanStack Table](https://tanstack.com/table/v8) and [TanStack Virtual](https://tanstack.com/virtual/v3), to manage much of the complexity surrounding the table implementation. This presents risk in the form of being able to quickly address any issues that arise. However, these libraries are active and eagerly support outside contribution, so we should be able to contribute as needed.
+
+The other primary risk is simply scope. This is a large, complex component, which can easily grow beyond our initial expectations, due to a variety of unforseen elements, such as missed important features, or misunderstood complexities.
+
+### Prior Art/Examples
+
+SLE Table
+
+![SLE table](./spec-images/sleTable.png)
+
+---
+
+## Design
+
+A user will be able to configure a `nimble-table` component fairly thoroughly through its markup. Including the columns to visualize, how to visualize a column, and in what order. A sketch of the markup should be similar to the following (in Angular):
+
+```html
+<nimble-table data="data">
+    <nimble-text-field-column
+        slot="columns"
+        columnId="firstName"
+    ></nimble-text-field-column>
+    <nimble-text-field-column
+        slot="columns"
+        columnId="lastName"
+    ></nimble-text-field-column>
+    <nimble-text-field-column
+        slot="columns"
+        columnId="profession"
+    ></nimble-text-field-column>
+</nimble-table>
+```
+
+_Placeholder for other design details as they are created_
+
+### API
+
+_Component Name_
+
+`nimble-table`
+
+_Feature APIs_
+
+The various APIs/features of the `nimble-table` will be split up amongst several different HLD documents. This section will serve to list them out and link to them as they become available:
+
+-   Data API :
+    -   Define what the interface is for setting/getting data on the table
```

Comment:
Perhaps this line should explicitly state that we need to consider the interface for our supported frameworks. Maybe something like this:

```suggestion
    -   Define what the interface is for setting/getting data on the table, taking into account any framework-specific binding concerns
```